### PR TITLE
Improve loading of `NXoff_geometry`

### PR DESCRIPTION
**Pull request by SimonHeybrock**
- Load vertices with vector3 dtype.
- Set field dims.

### CODE REVIEW

File: src/scippnexus/nxobject.py [hunk 1]
```diff
@@ -137,16 +137,34 @@ def _as_datetime(obj: Any):
     return None
 
 
+def _dtype_from_dataset(dataset: H5Dataset) -> sc.DType:
+    dtype = dataset.dtype
+    if str(dtype).startswith('str') or h5py.check_string_dtype(dtype):
```
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply src/scippnexus/nxobject.py:142]: This is just moved code, not a change here. I *think* it had to do with fixed- vs. variable-length strings (or whatever varieties HDF5 supports).

File: tests/nxoff_geometry_test.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import pytest
+import scipp as sc
+
+from scippnexus import NXentry, NXoff_geometry, NXroot
+
+
+@pytest.fixture()
+def nxroot(request):
+    """Yield NXroot containing a single NXentry named 'entry'"""
+    with h5py.File('dummy.nxs', mode='w', driver="core", backing_store=False) as f:
```
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply tests/nxoff_geometry_test.py:11]: You made this comment before, and as before: It *is* in-memory.

[the remark this answers is not in the document]
Comment (jl-wynen) [reply tests/nxoff_geometry_test.py:11]: right

File: src/scippnexus/nxobject.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jl-wynen) [reply src/scippnexus/nxobject.py:142]: ok

**Comment by SimonHeybrock** [comment]
For a bit of context, the actual shape group of DREAM now loads like this: 
image

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**